pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -372,6 +420,30 @@ func (p *PriorityQueue) backoffPod(pod *v1.Pod) {
 func (p *PriorityQueue) AddUnschedulableIfNotPresent(pod *v1.Pod) error {
 	p.lock.Lock()
 	defer p.lock.Unlock()
+
+	if util.EquivalenceClassEnabled() {
+		equivalenceClass := equivalence.NewClass(pod).AddPod(pod)
+		p.nominatedPods.add(pod, "")
+
+		if p.unschedulableQ.get(pod) != nil {
+			return fmt.Errorf("pod is already present in unschedulableQ")
+		}
+		if _, exists, _ := p.activeQ.Get(equivalenceClass); exists {
+			return fmt.Errorf("pod is already present in the activeQ")
+		}
+		// TODO (denkensk): Use equivalence class for backoff in follow up PR
+		if !p.receivedMoveRequest && isPodUnschedulable(pod) {
+			p.unschedulableQ.addOrUpdate(pod)
```

Comment:
The goal of this patch is to avoid "failure flood" as the PR described, it's not related to throughput or latency, performance issue are being fixed in the control panel path recently.